src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -421,19 +427,44 @@ async def build_pex(
     requirements_file_digest = EMPTY_DIGEST
     if request.requirements.file_path:
         argv.extend(["--requirement", request.requirements.file_path])
+
+        globs = PathGlobs(
+            [request.requirements.file_path],
+            glob_match_error_behavior=GlobMatchErrorBehavior.error,
+            description_of_origin=request.requirements.file_path_description_of_origin,
+        )
+
+        # use contents to invalidate here
+        requirements_file_digest_contents = await Get(
+            DigestContents,
+            PathGlobs,
+            globs,
+        )
+        metadata = read_lockfile_metadata(requirements_file_digest_contents[0].content)
+        if metadata.invalidation_digest != request.requirements.lockfile_hex_digest:
+            if python_setup.fail_on_invalid_lockfile:
+                raise ValueError("Invalid lockfile provided. [TODO: Improve message]")
+            else:
+                logger.warning("%s", "Invalid lockfile provided. [TODO: Improve message]")
+
         requirements_file_digest = await Get(
             Digest,
-            PathGlobs(
-                [request.requirements.file_path],
-                glob_match_error_behavior=GlobMatchErrorBehavior.error,
-                description_of_origin=request.requirements.file_path_description_of_origin,
-            ),
+            PathGlobs,
+            globs,
```

Comment:
FYI this is split over 3 lines because there was a trailing comma so Black forces it to be split. You can get back to one line by removing the comma and rerunning Black. 

You don't need to fix as that's what Black did, only an FYI. (can't remember if I've mentioned this already)

Reply:
👍